procsim/biomass/raw_product_generator.py

```diff
@@ -44,23 +43,18 @@ def _create_raw_product(self, dir_name, name_gen):
         self._logger.info('Create {}'.format(dir_name))
         full_dir_name = os.path.join(self._output_path, dir_name)
         os.makedirs(full_dir_name, exist_ok=True)
-        mph_file_name = os.path.join(dir_name, name_gen.generate_mph_file_name())
-        full_mph_file_name = os.path.join(self._output_path, mph_file_name)
+
+        bin_file_name = name_gen.generate_binary_file_name()
+        full_bin_file_name = os.path.join(full_dir_name, bin_file_name)
+        self._add_file_to_product(full_bin_file_name, self._size_mb)
+        
+        mph_file_name = name_gen.generate_mph_file_name()
+        full_mph_file_name = os.path.join(full_dir_name, mph_file_name)
         self._hdr.write(full_mph_file_name)
-        bin_file_name = os.path.join(dir_name, name_gen.generate_binary_file_name())
-        full_bin_file_name = os.path.join(self._output_path, bin_file_name)
-        self._generate_bin_file(full_bin_file_name, self._size_mb)
-        if self._zip_output:
-            self._zip_directory(full_dir_name, [full_mph_file_name, full_bin_file_name], [mph_file_name, bin_file_name])
 
-    def _zip_directory(self, dir_name: str, filenames: List[str], arcnames: List[str]):
-        # Note: Deletes input files afterwards
-        self._logger.debug('Archive to zip, extension {}'.format(self._zip_extension))
-        with zipfile.ZipFile(dir_name + self._zip_extension, 'w', compression=zipfile.ZIP_DEFLATED) as zipped:
-            for filename, arcname in zip(filenames, arcnames):
-                zipped.write(filename, arcname)
-            zipped.close()
-            shutil.rmtree(dir_name)
+        if self._zip_output:
+            shutil.make_archive(full_dir_name, 'zip', self._output_path, dir_name)
```

Comment:
Consider leaving code like this in it's own (static or global) method, despite of the fact that 'it's only two lines of code'. It's code with its own, isolated responsibility that could be re-used easily.

Reply:
The zip file MUST have self._zip_extension (user requirement). It must be possible to make the extension `.ZIP` (in captitals).

Reply:
I've moved this specific zip functionality to a more general static `zip_folder` function and have similarly generalized unzip functionality. Created archives should now always get renamed to use `self._zip_extension` and unzip functionality should not have a problem with this since it always tries to decompress the zip format. (e9b06c909cbf98a13d4e79d9a751c41a25167ee9)